boards/BMS/Src/App/App_CurrentSense.c

```diff
@@ -0,0 +1,104 @@
+#include <stddef.h>
+#include "App_CurrentSense.h"
+
+ExitCode App_CurrentSense_ConvertToMainCurrent1(
+    float  adc_voltage,
+    float *main_current_1)
+{
+    if (adc_voltage < 0.0f)
+        return EXIT_CODE_OUT_OF_RANGE;
+
+    if (main_current_1 == NULL)
+        return EXIT_CODE_INVALID_ARGS;
+
+    // HSNBV-D06 Output 1 (+/- 50A):
+    //
+    // +------------------+                +-------------+
+    // | HSNBV-D06        |---<1.1k>---+---| ADC Channel |
+    // | Output 1 Voltage |            |   +-------------+
+    // +------------------+            |
+    //                               <2.2k>
+    //                                 |
+    //                                ===
+    //                                GND
+    //
+    //             HSNBV-D06 Output 1 Voltage - Offset Voltage
+    // Current = ------------------------------------------------
+    //                             Sensitivity
+    //                                              1k + 2.2k
+    // HSNBV-D06 Output 1 Voltage = ADC Voltage x --------------
+    //                                                 2.2k
+    // Offset Voltage = 2.5V
+    //
+    // Sensitivity = 40mV/A
+
+    const float offset             = 2.5f;
+    const float sensitivity        = 40e-3f;
+    const float hsnbv_d06_output_1 = adc_voltage * (1e3f + 2.2e3f) / 2.2e3f;
+
+    *main_current_1 = (hsnbv_d06_output_1 - offset) / sensitivity;
+
+    return EXIT_CODE_OK;
+}
```

Comment:
Just reading through this, it really seems like a very hardware-specific thing. We can still test IO code if we explicitly include it in the gtest, so I don't see a particularly strong reason this shouldn't be in IO? I suppose the idea here is that if we change things like resistor values, we really shouldn't need to change `app` code.

Reply:
(Sorry for this being a belated comment, was tired when I did the first pass and just took another look at this after a full night of sleep)

Reply:
How can we add IO code to gtest since they include the STM32 headers? I wasn't aware testing IO code is an option.

Reply:
Oh they do? I thought in this case we would just not include any stm32 headers in that particular Io `.c/.h`? Then we could just include it in the files we build into the gtest target.

Reply:
Not all, but most Io code does include stm32 headers.

Did you mean we will just hard code exactly which Io .c/.h should be added to `gtest`?

Reply:
> Also do you mean we should just hard code exactly which Io .c/.h to add to gtest?

Yeah, that was basically what I was thinking.  Looking at this I think it would require one of:

1. adding an additional argument to `create_arm_binary_or_tests_for_board`
1. just putting the IO files in `GOOGLETEST_TEST_SRCS` 
1. just splitting the files we pass to `create_arm_binary_or_tests_for_board` based on whether or not we can compile them for x86 or not.

Personally I'm a fan of (3), since it means we aren't adding more arguments (as 1 does) and it doesn't feel like a bit of a hack (like 2 does).

Reply:
Do you mind writing some pseudo-code for option 3? Did you mean replacing the following arguments
```
    "${ARM_BINARY_APP_SRCS}"
    "${ARM_BINARY_APP_INCLUDE_DIRS}"
    "${ARM_BINARY_IO_SRCS}"
    "${ARM_BINARY_IO_INCLUDE_DIRS}"
```
with
```
    "${x86_COMPATIBLE_SRCS}" // This can be App or Io
    "${x86_COMPATIBLE_INCLUDE_DIRS}"  // This can be App or Io
    "${ARM_BINARY_SRCS}"  // This must be Io
    "${ARM_BINARY_INCLUDE_DIRS}" // This must be Io
```

Reply:
Yep basically